tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -40,121 +42,147 @@
+    #
+    _setup_lock = RLock()
+
+    def _setup(self) -> None:
         """
-        Cleans up any leftovers from the test suite run.
+        Prepares the test suite for execution (currently, it just builds the agent package)
+
+        Returns the path to the agent package.
         """
+        AgentTestSuite._setup_lock.acquire()
+
         try:
-            log.info("Removing %s", self.context.working_directory)
-            rmtree(self.context.working_directory.as_posix(), ignore_errors=True)
-        except:  # pylint: disable=bare-except
-            log.exception("Failed to cleanup the test run")
+            log.info("")
+            log.info("**************************************** [Build] ****************************************")
+            log.info("")
+            completed: Path = self.context.working_directory/"completed"
 
-    def _setup_node(self) -> None:
-        """
-        Prepares the remote node for executing the test suite.
-        """
-        agent_package_path = self._build_agent_package()
-        self._install_agent_on_node(agent_package_path)
+            if completed.exists():
+                log.info("Found %s. Build has already been done, skipping", completed)
+                return
+
+            log.info("Creating working directory: %s", self.context.working_directory)
+            self.context.working_directory.mkdir(parents=True)
+            self._build_agent_package()
 
-    def _build_agent_package(self) -> Path:
+            log.info("Completed setup, creating %s", completed)
+            completed.touch()
+
+        finally:
+            AgentTestSuite._setup_lock.release()
+
+    def _build_agent_package(self) -> None:
         """
         Builds the agent package and returns the path to the package.
         """
-        build_path = self.context.working_directory/"build"
-
-        log.info("Building agent package to %s", build_path)
+        log.info("Building agent package to %s", self.context.working_directory)
 
-        makepkg.run(agent_family="Test", output_directory=str(build_path), log=log)
+        makepkg.run(agent_family="Test", output_directory=str(self.context.working_directory), log=log)
 
-        package_path = build_path/"eggs"/f"WALinuxAgent-{AGENT_VERSION}.zip"
+        package_path: Path = self._get_agent_package_path()
         if not package_path.exists():
             raise Exception(f"Can't find the agent package at {package_path}")
 
-        log.info("Agent package: %s", package_path)
+        log.info("Built agent package as %s", package_path)
+
+    def _get_agent_package_path(self) -> Path:
+        """
+        Returns the path to the agent package.
+        """
+        return self.context.working_directory/"eggs"/f"WALinuxAgent-{AGENT_VERSION}.zip"
 
-        return package_path
+    def _clean_up(self) -> None:
+        """
+        Cleans up any leftovers from the test suite run. Currently just an empty placeholder for future use.
+        """
```

Comment:
The cleanup used to remove the working directory. Since this is part of the protocol enforced by self.execute(), I left it here as an empty method in case we need it in the future.